---
title: "ChIP-Seq Analysis"
author: "Cordell Browne"
date: "cordell@knights.ucf.edu"
output:
  html_document:
    highlight: pygments
    theme: cosmo
---

<br>
<br>

# {.tabset .tabset-fade}

## About the Data.1

ChIP-seq was used to analyze five breast cancer cell lines against the transcription factor ERa. Three of these cell lines are responsive to tamoxifen treatment, while two others are resistant to tamoxifen; there are eleven total sequenced libraries. For each sample, there is an associated peakset derived using the MACS peak caller. Only data for chromosome 18 was included in the data. The goal of the analysis is compare binding across samples and cell types. 


Tamoxifen is a nonsteroidal antiestrogen that has found successful applications for each stage of breast cancer in the treatment of selected patients. Tamoxifen was originally introduced for the treatment of advanced disease in postmenopausal women. The proven efficacy of tamoxifen and the low incidence of side effects made the drug an ideal agent to test as an adjuvant therapy for women with node-positive breast cancer. 


doi: 10.1016/0147-0272(92)90002-6. PMID: 1582240.

<br>
<br>
<br>

## About the Data.2 {.tabset}

Source: https://content.cruk.cam.ac.uk/bioinformatics/software/DiffBind/

```{r,include=FALSE}
# Load libraries 
library(kableExtra) 
library(DESeq2)
library(DiffBind)
library(ChIPseeker)
library(enrichplot)
library(ReactomePA)
library(clusterProfiler)
library(biomaRt)
library(org.Hs.eg.db)  
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
txdb <- TxDb.Hsapiens.UCSC.hg19.knownGene
```

```{r,include=FALSE}
#Get Data
setwd(file.path("./DiffBind_Vignette"))
samples <- read.csv("./tamoxifen.csv")
tamoxifen <- dba(sampleSheet ="./tamoxifen.csv") %>% 
  dba.blacklist() %>%
  dba.count()     %>%
  dba.normalize() %>%
  dba.contrast()  %>%
  dba.analyze()

#Display Dataframes
df.tamoxifen <- dba.show(tamoxifen)
tamoxifen.DB <- dba.report(tamoxifen)
tamoxifen.DB.DF <- as.data.frame(tamoxifen.DB)

#DESeq2 Object
tamoxifen.DESeq <- tamoxifen[["DESeq2"]]
dds.tamoxifen <-tamoxifen.DESeq[["DEdata"]]
ddseq <- DESeq(dds.tamoxifen)
res <- results(ddseq)
resDF <- as.data.frame(res)
```

### Samples
```{r,echo=FALSE}
kbl(samples)  %>%
  kable_paper() %>%
  scroll_box(width = "100%", height = "200px")
```

<br>
<br>
<br>


### BT474.1 Peaks
```{r,echo=FALSE}
kbl(tamoxifen[["peaks"]][[1]])  %>%
  kable_paper() %>%
  scroll_box(width = "100%", height = "200px")
```

<br>
<br>
<br>

## Qualitly Check

[MultiQC Sample File (_Not representitative of data_)](./multiqc_report.html)

<br>
<br>
<br>

## Pre-Analysis 

The following script takes files in /fastq folder ending in .fastq.gz and converts them into bam files 

```{r ,engine= 'bash',eval= FALSE}
#!/bin/bash

# set variables
INDEX= ./bowtie2/dm6
DATA_DIR= ./Data/ChIP-Seq


for f in ./fastq
 do
 
   base=$(basename $f .fastq.gz)
   echo $base
   echo $f
   echo $f >> bowtie_log.txt
   bowtie2 -x $INDEX -p 2 -U $DATA_DIR/$f 2>>bowtie_log.txt | samtools view -S -b -o $base.bam -
   echo "sorting $base.bam"
   samtools sort $base.bam -T $base -o $base.bam
   echo "indexing"
   samtools index $base.bam
   
done
```

Converting BAM to BED (need this for Differential Binding Analysis)

```{r, engine = 'bash', eval = FALSE}
cd ./directory

for x in *.bam ; do
    echo "print current:$x";
    bedtools bamTobed -i "$x" > "${x%.bam}.bed.gz";
done

echo "done"
```

Peak calling with MACS 2.0.

```{r, engine = 'bash', eval = FALSE}

for x in *.bam ; do
    echo "print current:$x";
    macs2 callpeak -t ${x} -f BAM -g mm -n ${x%.bam}_label -p 1e-3 --to-large;
done

echo "done"
```

<br>
<br>
<br>

# Differential Binding Analysis

<br>
<br>

__Multidimensional scaling (MDS)__ is a popular approach for graphically representing relationships between objects (e.g. plots or samples) in multidimensional space. Dimension reduction via MDS is achieved by taking the original set of samples and calculating a dissimilarity (distance) measure for each pairwise comparison of samples. The samples are then usually represented graphically in two dimensions such that the distance between points on the plot approximates their multivariate dissimilarity as closely as possible.

*I recommend to color by "tissue" and shape by "condition"*

```{r,echo=FALSE}
library(Glimma)
glimmaMDS(dds.tamoxifen, width=800, height=450)
```


## {.tabset .tabset-fade .tabset-pills}

<br>
<br>


__MA plots__ are a useful way to visualize the relationship between the overall binding level at each site and the magnitude of the change in binding enrichment between conditions, as well as the effect of normalization on data. An MA plot can be obtained for the Resistant vs Responsive contrast as follows:

Each point represents a binding site, with the 246 points in magenta representing sites identified as differentially bound. The bias towards enriched binding in the Responsive case (or loss of binding affinity in the Resistant case) can be visualized using MA and Volcano plots, as shown in the following Section.

### MA Plot i
```{r,echo=FALSE}
dba.plotMA(tamoxifen)
```

### MA Plot ii
```{r,echo=FALSE}
plotMA(res)
```

<br> 
<br>

##  {.tabset .tabset-fade .tabset-pills}

 __Volcano plots__ also highlight significantly differentially bound sites and show their fold changes. Here, however, the confidence statistic (FDR or p-value) is shown on a negative log scale, helping visualize the relationship between the magnitude of fold changes and the confidence that sites are differentially bound.


The plot shown, with the predominance of lower binding in the Resistant case evidenced by the greater number of significant sites on the negative side of the Fold Change (X) axis.

### Volcano Plot i

```{r,fig.width= 7,fig.height=7,echo=FALSE,message=FALSE}
library(EnhancedVolcano)
library(magrittr)
  EnhancedVolcano(res ,lab = rownames(res), x = 'log2FoldChange', y = 'pvalue')
```

<br>
<br>


### Volcano Plot ii

```{r,fig.width= 7,fig.height=7,echo=FALSE}
dba.plotVolcano(tamoxifen)
```

<br>
<br>

## {.tabset .tabset-fade .tabset-pills}

__Correlation heatmaps__

### Cell-Line Heatmap

```{r,fig.width= 10,fig.height=7,echo=FALSE}
dba.plotHeatmap(tamoxifen)
```
The plot shows that while the replicates for each cell line cluster together appropriately, the cell lines do not cluster into groups corresponding to those that are responsive (MCF7, T47D, and ZR75) vs. those resistant (BT474 and MCF7r) to tamoxifen treatment. It also shows that the two most highly correlated cell lines are the two MCF7-based ones, even though they respond differently to tamoxifen treatment.

<br>
<br>
<br>


### Binding Site Heatmap

Another way to view the patterns of binding affinity directly in the differentially bound sites is via a binding affinity heatmap, showing the read scores for some or all of the binding sites. This can be plotted for the example case as follows:

```{r,fig.width= 10,fig.height=7,echo=FALSE}
hmap <- colorRampPalette(c("blue", "black", "red"))(n = 13)
readscores <- dba.plotHeatmap(tamoxifen, contrast=1, correlations=FALSE,
                              scale="row", colScheme = hmap)
```

the plot shows the affinities and clustering of the differentially bound sites (rows), as well as the sample clustering (columns).

<br>
<br>
<br>

## II.

The following venn diagram shows overlaps between peaksets.
```{r,echo=FALSE,results='hide',fig.keep='all'}
dba.plotVenn(tamoxifen, contrast=1, bDB=TRUE, bGain=TRUE, bLoss=TRUE, bAll=TRUE)
```


### __ChIP-seq Profiling__ {.tabset .tabset-fade}

#### BT474

```{r,echo=FALSE,fig.width=7,message=FALSE,warning=FALSE,results='hide',fig.keep='all'}
peaks.consensus <- dba.peakset(tamoxifen, bRetrieve = T)

# extracting HeLA peaks
peaks.BT474_rep1 <- peaks.consensus[tamoxifen$called[,1]==1] # peaks called in rep 1
peaks.BT474_rep2 <- peaks.consensus[tamoxifen$called[,2]==1] # peaks called in rep 2

# adding an unified affinity scores column (re-formatting data)
peaks.BT474_rep1$Score <- peaks.BT474_rep1$BT4741
peaks.BT474_rep2$Score <- peaks.BT474_rep2$BT4742

# creating genomicRangesList object holding replicates 1 and 2
grL.BT474<- GRangesList(BT474_rep1=peaks.BT474_rep1, BT474_rep2=peaks.BT474_rep2)

# plotting using affinity scores as a weight for peaks height
covplot(grL.BT474 , weightCol = "Score", title = "ChIP Peaks Over Chr. 18")

#x<-as.data.frame(tamoxifen$called)
```

In these images we compare peaks across replicates particularly BT474. This should give us visual assessment of variability between replicates: peaks locations and strength should match in an idealistic scenario

From the figures provided it can be concluded which pair of replicates is most consistent.

<br>
<br>

#### MCF7

```{r,echo=FALSE,fig.width=7,message=FALSE,warning=FALSE,results='hide',fig.keep='all'}
# extracting peaks
peaks.MCF7_rep1 <- peaks.consensus[tamoxifen$called[,3]==1] 
peaks.MCF7_rep2 <- peaks.consensus[tamoxifen$called[,4]==1] 
peaks.MCF7_rep3 <- peaks.consensus[tamoxifen$called[,5]==1]
# adding an unified affinity scores column (re-formatting data)
peaks.MCF7_rep1$Score <- peaks.MCF7_rep1$MCF71
peaks.MCF7_rep2$Score <- peaks.MCF7_rep2$MCF72
peaks.MCF7_rep3$Score <- peaks.MCF7_rep3$MCF73
# creating genomicRangesList object
grL.MCF7<- GRangesList(MCF7_rep1=peaks.MCF7_rep1, MCF7_rep2=peaks.MCF7_rep2, MCF7_rep3=peaks.MCF7_rep3)
# plotting using affinity scores as a weight for peaks height
covplot(grL.MCF7 , weightCol = "Score", title = "ChIP Peaks Over Chr. 18")
```

<br>
<br>

#### T47D

```{r,echo=FALSE,fig.width=7,message=FALSE,warning=FALSE,results='hide',fig.keep='all'}
peaks.consensus <- dba.peakset(tamoxifen, bRetrieve = T)

# extracting peaks
peaks.T47D_rep1 <- peaks.consensus[tamoxifen$called[,6]==1] # peaks called in rep 1
peaks.T47D_rep2 <- peaks.consensus[tamoxifen$called[,7]==1] # peaks called in rep 2

# adding an unified affinity scores column 
peaks.T47D_rep1$Score <- peaks.T47D_rep1$T47D1
peaks.T47D_rep2$Score <- peaks.T47D_rep2$T47D2

# creating genomicRangesList object 
grL.T47D<- GRangesList(T47D_rep1=peaks.T47D_rep1, T47D_rep2=peaks.T47D_rep2)

# plotting using affinity scores as a weight for peaks height
covplot(grL.T47D , weightCol = "Score", title = "ChIP Peaks Over Chr. 18")
```

<br>
<br>

#### ZR75

```{r,echo=FALSE,fig.width=7,message=FALSE,warning=FALSE,results='hide',fig.keep='all'}
peaks.consensus <- dba.peakset(tamoxifen, bRetrieve = T)

# extracting HeLA peaks
peaks.ZR75_rep1 <- peaks.consensus[tamoxifen$called[,10]==1] # peaks called in rep 1
peaks.ZR75_rep2 <- peaks.consensus[tamoxifen$called[,11]==1] # peaks called in rep 2

# adding an unified affinity scores column (re-formatting data)
peaks.ZR75_rep1$Score <- peaks.ZR75_rep1$ZR751
peaks.ZR75_rep2$Score <- peaks.ZR75_rep2$ZR752

# creating genomicRangesList object holding replicates 1 and 2
grL.ZR75<- GRangesList(ZR75_rep1=peaks.ZR75_rep1, ZR75_rep2=peaks.ZR75_rep2)

# plotting using affinity scores as a weight for peaks height
covplot(grL.ZR75 , weightCol = "Score", title = "ChIP Peaks Over Chr. 18")

```

<br>
<br>


### __Heatmap of ChIP peaks among different cell lines__

```{r,echo=FALSE,message=FALSE,warning=FALSE,results='hide',fig.keep='all'}
# extracting peaks for each cell line present across replicates
peaks.BT474 <- peaks.consensus[tamoxifen$called[,1]==1 & tamoxifen$called[,2]==1]
peaks.MCF7 <- peaks.consensus[tamoxifen$called[,3]==1 & tamoxifen$called[,4]==1 
  & tamoxifen$called[,5]==1 & tamoxifen$called[,8]==1 & tamoxifen$called[,9]==1]
peaks.T47D <- peaks.consensus[tamoxifen$called[,6]==1 & tamoxifen$called[,7]==1]
peaks.ZR75 <- peaks.consensus[tamoxifen$called[,10]==1 & tamoxifen$called[,11]==1]

# getting TSS regions
promoter <- getPromoters(TxDb=txdb, upstream=3000, downstream=3000)

# calculating tagMatrix
tagMatrix.1 <- getTagMatrix(peaks.BT474, windows=promoter)
tagMatrix.2 <- getTagMatrix(peaks.MCF7, windows=promoter)
tagMatrix.3 <- getTagMatrix(peaks.T47D, windows=promoter)
tagMatrix.4 <- getTagMatrix(peaks.ZR75, windows=promoter)

# preparing tagMatrix list to enable cell lines comparisions
tagMatrixList <- list(BT474=tagMatrix.1, MCF7=tagMatrix.2, T47D=tagMatrix.3, ZR75=tagMatrix.4)

# plotting tagMatrix heatmaps for each cell line
tagHeatmap(tagMatrixList, xlim=c(-3000, 3000), color=NULL)
```

<br>
<br>

### __Average profile of ChIP peaks among different cell lines__

```{r,echo=FALSE,message=FALSE,warning=FALSE,results='hide',fig.keep='all'}
# plotting average profile of ChIP peaks among different cell lines
plotAvgProf(tagMatrixList, xlim=c(-3000, 3000))
```

I defined the TSS region, by default set to -3kb to 3kb. The output of is shown above.

<br>
<br>

### __Genomics locations: dataset comparisons with barplot__ 

```{r,echo=FALSE,message=FALSE,warning=FALSE,results='hide',fig.keep='all'}
# annotating peaks
peaks.BT474_ann <- annotatePeak(peaks.BT474, tssRegion=c(-3000, 3000), TxDb=txdb, annoDb="org.Hs.eg.db")
peaks.MCF7_ann <- annotatePeak(peaks.MCF7, tssRegion=c(-1000, 2000), TxDb=txdb, annoDb="org.Hs.eg.db")
peaks.T47D_ann <- annotatePeak(peaks.T47D, tssRegion=c(-1000, 2000), TxDb=txdb, annoDb="org.Hs.eg.db")
peaks.ZR75_ann <- annotatePeak(peaks.ZR75, tssRegion=c(-1000, 2000), TxDb=txdb, annoDb="org.Hs.eg.db")

# creating list holding annotations for different cell lines
list.annotations <- list(BT474=peaks.BT474_ann, MCF7=peaks.MCF7_ann, HT47D=peaks.T47D_ann, ZR75=peaks.ZR75_ann)
# creating barplot for HeLa, neural, HepG2 and sknsh peaks genomic annotations
plotAnnoBar(list.annotations)
#vennpie(peaks.BT474_ann)
```

<br>
<br>

### __Distance to TSS Comparisons__

```{r,echo=FALSE,message=FALSE,results='hide',fig.keep='all'}

# plotting distance to TSS for all cell lines in our annotation list
plotDistToTSS(list.annotations)
```

<br>
<br>
<br>

# Functional Analysis In Progress ... 


```{r,eval = FALSE}
#biomart
ensembl <- useEnsembl(biomart = "genes", dataset = "hsapiens_gene_ensembl",version = 95)
# running query: extracting ENTREZID 
genes.chr18 <- getBM(attributes= "entrezgene", filters=c("chromosome_name"), values=18, mart=ensembl)
# reformatting output to character string 
genes.universe <- as.character(as.numeric(as.matrix(genes.chr18)))
```
```{r,eval = FALSE}
#1 selecting annotated peaks for functional enrichment 
data.peaks_ann <- peaks.MCF7_ann
#x <- as.data.frame(data.peaks_ann)
#1 finding enriched Reactome pathways using chromosome 
pathway.reac1 <- enrichPathway(as.data.frame(data.peaks_ann)$geneId, universe = genes.universe)
#2 selecting peaks
data.peaks <- peaks.T47D
#2  seq2gene function for many-to-many mapping based on sequence regions 
genes.m2m <- seq2gene(data.peaks, tssRegion = c(-1000, 1000), flankDistance = 3000, TxDb=txdb)
#2 finding enriched Reactome pathways 
pathway.reac2 <- enrichPathway(genes.m2m, universe = genes.universe)
#2 visualise enrichment results
dotplot(pathway.reac2)

```
```{r,eval = FALSE}
# creating a gene list with ENTREZID 
list.genes = lapply(list.annotations, function(i) as.data.frame(i)$geneId)
names(list.genes) <- sub("_", "\n", names(list.genes))
# running enrichedGO 
compMF <- compareCluster(geneCluster = list.genes,fun= "enrichGO",
                       pvalueCutoff  = 0.05, pAdjustMethod = "BH",
                       OrgDb='org.Hs.eg.db', ont="MF")
# results dotplot
dotplot(compMF)
# simplify results 
compMF.flr <- simplify(compMF, cutoff = 0.7, by = "p.adjust", select_fun = min, measure = "Wang", semData = NULL)
#dotplot GO terms
dotplot(compMF.flr)
```


# {.tabset}

## Citation

- G Yu, LG Wang, QY He. ChIPseeker: an R/Bioconductor package for ChIP peak annotation, comparison and visualization. Bioinformatics 2015, 31(14):2382-2383. doi:[10.1093/bioinformatics/btv145](http://dx.doi.org/10.1093/bioinformatics/btv145)

- C.S. Ross-Innes, R. Stark, A.E. Teschendorff, K.A. Holmes, H.R. Ali, M.J. Dunning, G.D. Brown, O. Gojis, I.O. Ellis, A.R. Green, et al. Differential oestrogen receptor binding is associated with clinical outcome in breast cancer. Nature, 481(7381):389–393, 2012.

- Haley M. Amemiya, Anshul Kundaje, and Alan P. Boyle. The ENCODE blacklist: Identification of problematic regions of the genome. Scientific Reports, 9(1):9354, 2019.

- Gord Brown. GreyListChIP: Grey lists–mask artefact regions based on ChIP inputs, 2015. URL: https://bioconductor.org/packages/GreyListChIP/, doi:10.18129/B9.bioc.GreyListChIP.

- https://nbisweden.github.io/workshop-archive/workshop-ChIP-seq/2018-11-07/labs/lab-ChIPseeker.html#Profile


## Session Info 

```{r}
# Print session info
sessioninfo::session_info()
```

