---
title: "ChIP-Seq Analysis"
author: "Cordell Browne"
date: "cordell@knights.ucf.edu"
output:
  html_document:
    theme: cosmo
---

# Introduction {.tabset .tabset-fade .tabset-pills}

## Objective 

The data for ChIP-seq peak calling are stacks of aligned reads across a genome. Some of these stacks correspond to the signal of interest (e.g. binding of a transcription factor or modified histone). Many other stacks are regarded as molecular or experimental noise, or as being influenced by a systematically greater accessibility of measurement by the experiment at that particular genomic location. The goal of this experiment is to understand how our transcription factors compare to one another in regards to binding sites.

## About the Data

Source: https://content.cruk.cam.ac.uk/bioinformatics/software/DiffBind/DiffBind_vignette_data.tar.gz

```{r,include=FALSE}
# Load libraries 
library(DESeq2)
library(DiffBind)
library(ChIPseeker)
library(ReactomePA)
library(clusterProfiler)
library(biomaRt)
library(org.Hs.eg.db)  
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
txdb <- TxDb.Hsapiens.UCSC.hg19.knownGene
```
The easiest way to set up an experiment to analyze is with a sample sheet. The sample sheet can be a , or it can be read directly from a csv file. Here is the example sample sheet read into a from a csv file:

The peaksets are read in using the following DiffBind function:
```{r,include=FALSE}
#Get Data
setwd(file.path("./DiffBind_Vignette"))
samples <- read.csv("./tamoxifen.csv")
tamoxifen <- dba(sampleSheet ="./tamoxifen.csv") %>% 
  dba.blacklist() %>%
  dba.count()     %>%
  dba.normalize() %>%
  dba.contrast()  %>%
  dba.analyze()

#Display Dataframes
df.tamoxifen <- dba.show(tamoxifen)
tamoxifen.DB <- dba.report(tamoxifen)
tamoxifen.DB.DF <- as.data.frame(tamoxifen.DB)

#DESeq2 Object
tamoxifen.DESeq <- tamoxifen[["DESeq2"]]
dds.tamoxifen <-tamoxifen.DESeq[["DEdata"]]
ddseq <- DESeq(dds.tamoxifen)
res <- results(ddseq)
resDF <- as.data.frame(res)
```

# Differential Binding Analysis

## PCA 


While the correlation heatmaps already seen are good for showing clustering, plots based on principal components analysis can be used to give a deeper insight into how samples are associated. A PCA plot corresponding to Figure 2, which includes normalized read counts for all 2845 binding sites, can be obtained as follows:

The resulting plot (Figure 5) shows all the MCF7-derived samples (red) clustering on one side of the first (horizontal) component, with the Responsive and Resistant samples not separable either in the first nor in the second (vertical) component.3

*I recommend to color by "tissue" and shape by "condition"*

```{r}
library(Glimma)
glimmaMDS(dds.tamoxifen, width=800, height=450)
```

The bias towards enriched binding in the Responsive case (or loss of binding affinity in the Resistant case) can be visualized using MA and Volcano plots, as shown in the following Section.
```{r}
dba.plotVenn(tamoxifen, contrast=1, bDB=TRUE, bGain=TRUE, bLoss=TRUE, bAll=FALSE)
```
Venn diagrams are also useful for examining overlaps between peaksets, particularly when determining how best to derive consensus peaksets for read counting and further analysis. Section 8, which discusses consensus peaksets, shows a number of Venn plots in context, and the help page for dba.plotVenn has a number of additional examples.

## MA Plots {.tabset .tabset-fade .tabset-pills}
MA plots are a useful way to visualize the relationship between the overall binding level at each site and the magnitude of the change in binding enrichment between conditions, as well as the effect of normalization on data. An MA plot can be obtained for the Resistant vs Responsive contrast as follows:

### MA Plot i
```{r}
dba.plotMA(tamoxifen)
```

### MA Plot ii
```{r}
plotMA(res)
```


The plot is shown in Figure 7. Each point represents a binding site, with the 246 points in magenta representing sites identified as differentially bound. There is a blue horizontal line through the origin (0 LFC), as well as a horizontal red curve representing a non-linear loess fit showing the underlying relationship between coverage levels and fold changes. The plot shows how the differentially bound sites appear to have a minimum absolute log fold difference of somewhere between one and two. As we have already seen, it also shows that more ERa binding sites lose binding affinity in the tamoxifen resistant condition than gain binding affinity, as evidenced by more red dots below the center line than are above it. This same data can also be shown with the concentrations of each sample groups plotted against each other plot using dba.plotMA(tamoxifen, bXY=TRUE).
Section 7 contains several examples of MA plots, including showing non-normalized data, and the ability to plot any subset of samples against any other set of sample.

## Volcano Plots {.tabset .tabset-fade .tabset-pills}

Similar to MA plots, Volcano plots also highlight significantly differentially bound sites and show their fold changes. Here, however, the confidence statistic (FDR or p-value) is shown on a negative log scale, helping visualize the relationship between the magnitude of fold changes and the confidence that sites are differentially bound.
For example, the same data as plotted in Figure 7 can be visualized as a volcano plot:

### Volcano Plot i

```{r,fig.width= 7,fig.height=7}
library(EnhancedVolcano)
library(magrittr)
  EnhancedVolcano(res ,lab = rownames(res), x = 'log2FoldChange', y = 'pvalue')
```


### Volcano Plot ii

```{r,fig.width= 7,fig.height=7}
dba.plotVolcano(tamoxifen)
```

The plot is shown in Figure 8, with the predominance of lower binding in the Resistant case evidenced by the greater number of significant sites on the negative side of the Fold Change (X) axis.


## Heatmaps {.tabset .tabset-fade .tabset-pills}

The effect of different scoring methods (normalization) can be examined in these plots by setting the score parameter to a different value. The default value, DBA_SCORE_NORMALIZED, uses the normalized read counts (see Section 7). Another scoring method is to use RPKM fold (RPKM of the ChIP reads divided by RPKM of the control reads); a correlation heatmap for all the data using this scoring method can be obtained by typing dba.plotHeatmap(tamoxifen, score=DBA_SCORE_RPKM_FOLD).

### Cell-Line Heatmap

```{r,fig.width= 10,fig.height=7}
dba.plotHeatmap(tamoxifen)
```
The resulting plot (Figure 1) shows that while the replicates for each cell line cluster to- gether appropriately, the cell lines do not cluster into groups corresponding to those that are responsive (MCF7, T47D, and ZR75) vs. those resistant (BT474 and MCF7r) to ta- moxifen treatment. It also shows that the two most highly correlated cell lines are the two MCF7-based ones, even though they respond differently to tamoxifen treatment.

### Binding Site Heatmape
Another way to view the patterns of binding affinity directly in the differentially bound sites is via a binding affinity heatmap, showing the read scores for some or all of the binding sites. This can be plotted for the example case as follows:

```{r,fig.width= 10,fig.height=7}
hmap <- colorRampPalette(c("red", "black", "green"))(n = 13)
readscores <- dba.plotHeatmap(tamoxifen, contrast=1, correlations=FALSE,
                              scale="row", colScheme = hmap)
```
Figure 10 shows the affinities and clustering of the differentially bound sites (rows), as well as the sample clustering (columns). In this case, the (normalized) counts have been row scaled, and a red/green heatmap color palette applied.

# ChIP-seq downstream analysis

```{r}
peaks.consensus <- dba.peakset(tamoxifen, bRetrieve = T)

# extracting HeLA peaks
peaks.HeLa_rep1 <- peaks.consensus[tamoxifen$called[,1]==1] # peaks called in rep 1
peaks.HeLa_rep2 <- peaks.consensus[tamoxifen$called[,2]==1] # peaks called in rep 2

# adding an unified affinity scores column (re-formatting data)
peaks.HeLa_rep1$Score <- peaks.HeLa_rep1$BT4741
peaks.HeLa_rep2$Score <- peaks.HeLa_rep2$BT4742

# plotting coverage for replicate 1, using affinity scores as a weight for peaks height
covplot(peaks.HeLa_rep1, weightCol = "Score",chrs=c("chr1", "chr18"))

# zooming in to a selected region is also possible
covplot(peaks.HeLa_rep1, weightCol = "Score", xlim=c(0, 1e07))
```

```{r}
# creating genomicRangesList object holding replicates 1 and 2
grL.HeLa <- GRangesList(HeLa_rep1=peaks.HeLa_rep1, HeLa_rep2=peaks.HeLa_rep2)

# plotting using affinity scores as a weight for peaks height
covplot(grL.HeLa, weightCol = "Score")

# zooming in
covplot(grL.HeLa)
```

```{r}
x<-as.data.frame(tamoxifen$called)
```

```{r}

```

```{r}

```

```{r}

```

```{r}

```

```{r}

```

```{r}

```

```{r}

```

```{r}

```

```{r}

```



