---
title: "Functional Enrichment Analysis "
author: "Cordell Browne"
date: "cordell@knights.ucf.edu"
output:
  html_document:
    theme: cosmo
    toc: yes
    toc_float: yes
---

<br>

# Enrichr

Enrichment analysis is a popular method for analyzing gene sets generated by genome-wide experiments. Enrichr currently contains a large collection of diverse gene set libraries available for analysis and download.  Overall, Enrichr is a comprehensive resource for curated gene sets and a search engine that accumulates biological knowledge for further biological discoveries. 


Enrichr is freely available at: http://amp.pharm.mssm.edu/Enrichr.

https://cran.r-project.org/web/packages/enrichR/vignettes/enrichR.html


```{r,echo=FALSE}
library(enrichR)
library(plotly)
setEnrichrSite("Enrichr")
#
websiteLive <- TRUE
#
dbs <- listEnrichrDbs()
```

```{r,message=FALSE,echo=FALSE}
#1.Ontologies
#2. Pathways
#3. Diseases/Drugs
#4. Cell Types
#5. Transcription


dbs <- c("GO_Molecular_Function_2015", "GO_Cellular_Component_2015", "GO_Biological_Process_2015","Human_Phenotype_Ontology",
         "BioPlanet_2019","WikiPathway_2021_Human", "KEGG_2021_Human", 
         "DisGeNET","PheWeb_2019",
         "Human_Gene_Atlas","Descartes_Cell_Types_and_Tissue_2021",
         "ChEA_2016","ENCODE_and_ChEA_Consensus_TFs_from_ChIP-X"
         
         )

genes <- c('RPSAP47', 'WIF1', 'KRT36', "NTS","LAMB4","SHOX",
           'CFTR',  'KRTAP3-1', 'RMST', 'C9orf152', 'KRT36',
           "HMGCS2","TNN","SHISA9","MAB21L2")

if (websiteLive) {
    enriched <- enrichr(genes, dbs)
}
```

<br>

## Ontologies {.tabset .tabset-fade}

<br>
The top 9 results from each of the databases of interest relating to gene ontology are displayed below.
<br>

A common practice that is arguably quite reasonable, albeit not entirely devoid of shortcomings, is to: (a) apply an adjusted p-value ("q-value") cutoff of 0.01-0.1 to filter enriched terms, (b) use the "combined score" output (which has been extensively shown to outperform other ranking metrics due to its inherent z-score permutation background correction on Fisher's exact test p-value1) to sort those filtered terms, in descending order, and then (c) select top X highest ranked terms to identify significantly over-represented functional categories.

### Molecular Function
```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[1]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[1], yaxis= yaxis, margin= list(t= 100))

fig
```

(1) Bicarbonate transport proteins facilitate the movement of membrane-impermeant HCO3− across membranes to accelerate disposal of waste CO2, control cellular and whole-body pH, and to regulate fluid movement and acid/base secretion. Their high metabolic rate make them of interest in squamous cell carcinomas .
(2) Chloride channels is associated with CFTR is a cyclic AMP (cAMP)-activated chloride (Cl-) channel and a regulator of outwardly rectifying Cl- channels (ORCCs) in airway epithelia (CFTR is discussed further in this analysis).


1. https://doi.org/10.1042/BJ20081634

2. https://doi.org/10.1073/pnas.95.5.2674



### Cellular Component
```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[2]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[2], yaxis= yaxis, margin= list(t= 100))

fig
```

(1) Alpha-amino-3-hydroxy-5-methyl-4-isoxazolepropionic acid (AMPA) receptors are responsible for the bulk of fast excitatory synaptic transmission throughout the CNS and their modulation is the ultimate mechanism that underlies much of the plasticity of excitatory transmission that is expressed in the brain. Increasing the post-synaptic response to a stimulus is achieved either through increasing the number of AMPA receptors at the post-synaptic surface or by increasing the single channel conductance of the receptors expressed. This was shown to be the basis of LTP mechanisms. More recently, it has become clear that these two apparently different mechanisms to increase AMPA receptor response can be combined, with the insertion of Ca2+ permeable AMPA receptors that have a high conductance in response to tetanic stimulation. Thus the overall conductance of a synapse may be increased by receptor insertion rather than modification of existing receptors.

(2) Intermediate filament (IF) proteins are traditionally known for their roles in fostering structural integrity in cells and tissues. In cancer, IF proteins serve as diagnostic markers, as tumor cells partially retain their original signature expression of IF proteins.The evidence already in hand establishes that IF proteins function beyond their classical roles as markers and serve as effectors of tumorgenesis

1. https://doi.org/10.1073/pnas.0611698104

2. https://doi.org/10.3390/cells8050497

### Biological Process
```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[3]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[3], yaxis= yaxis, margin= list(t= 100))

fig
```



### Human Phenotype
```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[4]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[4], yaxis= yaxis, margin= list(t= 100))

fig
```



## Pathways {.tabset .tabset-fade}

<br>
The top 9 results from each of the databases of interest relating to molecular pathways are displayed below.
<br>

### BioPlanet 2019
```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[5]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[5], yaxis= yaxis, margin= list(t= 100))

fig
```

(1) Ketone body metabolism was described before in ontolgoies their seems that some pancreatic disorders are overlapping with our disease state. 

(2) CFTR  is responsible for regulating the proper flow of chloride and sodium (a component of salt) in and out of the cell membranes in the lungs and other organs 

(3) The PITX2 gene provides instructions for making a protein that attaches (binds) to specific regions of DNA and regulates the activity of other genes. 

(4) Inactivation of GSK3  by Akt causes accumulation of beta-catenin in alveolar macrophages. GSK3 has been implicated in the development of insulin resistance, primarily based on its role in regulation of glycogen synthesis. However, GSK3 is involved in many other important signaling cascades which may regulate glucose homeostasis and the development of insulin resistance.Lrp5/β-Catenin Signaling Controls Lung Macrophage Differentiation and Inhibits Resolution of Fibrosis. This seems to be relating to CFTR.

1. https://pubmed.ncbi.nlm.nih.gov/9405449/

2. https://medlineplus.gov/genetics/gene/pitx2/

3. https://medlineplus.gov/genetics/gene/pitx2/

4. https://doi.org/10.1016/j.diabres.2007.01.033


### WikiPathway 2021 Human
```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[6]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[6], yaxis= yaxis, margin= list(t= 100))

fig
```

Again, this brings up more Ketone body metabolism relationships correlated with our gene set.

###  KEGG 2021 Human
```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[7]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[7], yaxis= yaxis, margin= list(t= 100))

fig
```


## Diseases/Drugs {.tabset .tabset-fade}

<br>
The top 9 results from each of the databases of interest relating to diseases and drugs are displayed below.
<br>

### DisGeNET
```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[8]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[8], yaxis= yaxis, margin= list(t= 100))

fig
```

(1) Chloasma is a required hypermelanosis of sun-exposed areas occurred during pregnancy and it can affect 50-70% of pregnant women.

(2) There are a variety of disorders of the pancreas (not diabetes) including acute pancreatitis, chronic pancreatitis, hereditary pancreatitis, and pancreatic cancer.


1. https://pubmed.ncbi.nlm.nih.gov/19140277/


### PheWeb 2019

```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[9]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[9], yaxis= yaxis, margin= list(t= 100))

fig
```



## Cell Types {.tabset .tabset-fade}

<br>
The top 9 results from each of the databases of interest relating to cell type are displayed below.
<br>

### Human Gene Atlas

```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[10]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[10], yaxis= yaxis, margin= list(t= 100))

fig
```

Data is seemingly related to the GI tract .

### Descartes Cell Types and Tissue 2021

```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[11]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[11], yaxis= yaxis, margin= list(t= 100))

fig
```



## Transcription {.tabset .tabset-fade}

<br>
The top 9 results from each of the databases of interest relating to transcription are displayed below.
<br>

### ChEA 2016

```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[12]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[12], yaxis= yaxis, margin= list(t= 100))

fig
```

(1) TRIM28 (Tripartite Motif Containing 28) is a Protein Coding gene. Diseases associated with TRIM28 include Wilms Tumor 1 and Renal Wilms' Tumor (Childhood cancer that starts in the kidneys). Among its related pathways are Gene Expression and E2F transcription factor network. Gene Ontology (GO) annotations related to this gene include _RNA binding and sequence-specific DNA binding_. 

(2) IRF1 (Interferon Regulatory Factor 1) is a Protein Coding gene. Diseases associated with IRF1 include Gastric Cancer and Lung Cancer. Among its related pathways are Immune response IFN gamma signaling pathway and Bacterial infections in CF airways. Gene Ontology (GO) annotations related to this gene include _DNA-binding transcription factor activity and RNA polymerase II cis-regulatory region sequence-specific DNA binding_.

1. https://www.genecards.org/cgi-bin/carddisp.pl?gene=TRIM28&keywords=trim28
2. https://www.genecards.org/cgi-bin/carddisp.pl?gene=IRF1&keywords=irf1

### ENCODE and ChEA Consensus TFs from ChIP-X"

```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[13]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[13], yaxis= yaxis, margin= list(t= 100))

fig
```
(1) PPARD (Peroxisome Proliferator Activated Receptor Delta) is a Protein Coding gene. Diseases associated with PPARD include Diabetic Cataract and Abdominal Obesity-Metabolic Syndrome 1. Among its related pathways are Metabolism and Pyruvate metabolism and Citric Acid (TCA) cycle. Gene Ontology (GO) annotations related to this gene include _DNA-binding transcription factor activity and transcription coactivator activity_. 


(2) TP53  encodes a tumor suppressor protein containing transcriptional activation, DNA binding, and oligomerization domains. The encoded protein responds to diverse cellular stresses to regulate expression of target genes, thereby inducing cell cycle arrest, apoptosis, senescence, DNA repair, or changes in metabolism. Mutations in this gene are associated with a variety of human cancers, including hereditary cancers such as Li-Fraumeni syndrome. TP53 mutations are universal across cancer types. The loss of a tumor suppressor is most often through large deleterious events, such as frameshift mutations, or premature stop codons. 


(3) YY1 (YY1 Transcription Factor) is a Protein Coding gene. Diseases associated with YY1 include Gabriele-De Vries Syndrome and Insulinoma. Among its related pathways are mTOR signaling pathway and Sterol Regulatory Element-Binding Proteins (SREBP) signalling. Gene Ontology (GO) annotations related to this gene include _DNA-binding transcription factor activity and transcription coactivator activity_. 

1. https://www.genecards.org/cgi-bin/carddisp.pl?gene=PPARD
2. https://www.genecards.org/cgi-bin/carddisp.pl?gene=TP53&keywords=tp53
3. https://www.genecards.org/cgi-bin/carddisp.pl?gene=YY1&keywords=yy1

