---
title: "Gene Expression Correlation"
author: "Cordell Browne"
date: "4/9/2022"
output:
  html_document:
    theme: cosmo
    toc: yes
    toc_float: yes
---

<br>

# Enrichr

Enrichment analysis is a popular method for analyzing gene sets generated by genome-wide experiments. Enrichr currently contains a large collection of diverse gene set libraries available for analysis and download.  Overall, Enrichr is a comprehensive resource for curated gene sets and a search engine that accumulates biological knowledge for further biological discoveries. 


Enrichr is freely available at: http://amp.pharm.mssm.edu/Enrichr.

https://cran.r-project.org/web/packages/enrichR/vignettes/enrichR.html


```{r,echo=FALSE}
library(enrichR)
library(plotly)
setEnrichrSite("Enrichr")
#
websiteLive <- TRUE
#
dbs <- listEnrichrDbs()
```

```{r,message=FALSE,echo=FALSE}
#1.Ontologies
#2. Pathways
#3. Diseases/Drugs
#4. Cell Types
#5. Transcription


dbs <- c("GO_Molecular_Function_2015", "GO_Cellular_Component_2015", "GO_Biological_Process_2015","Human_Phenotype_Ontology",
         "BioPlanet_2019","WikiPathway_2021_Human", "KEGG_2021_Human", 
         "DisGeNET","PheWeb_2019",
         "Human_Gene_Atlas","Descartes_Cell_Types_and_Tissue_2021",
         "ChEA_2016","ENCODE_and_ChEA_Consensus_TFs_from_ChIP-X"
         
         )

genes <- c('RPSAP47', 'WIF1', 'KRT36', "NTS","LAMB4","SHOX",
           'CFTR',  'KRTAP3-1', 'RMST', 'C9orf152', 'KRT36')

if (websiteLive) {
    enriched <- enrichr(genes, dbs)
}
```

<br>

## Ontologies {.tabset .tabset-fade}

<br>
The top 9 results from each of the databases of interest relating to gene ontology are displayed below.
<br>

A common practice that is arguably quite reasonable, albeit not entirely devoid of shortcomings, is to: (a) apply an adjusted p-value ("q-value") cutoff of 0.01-0.1 to filter enriched terms, (b) use the "combined score" output (which has been extensively shown to outperform other ranking metrics due to its inherent z-score permutation background correction on Fisher's exact test p-value1) to sort those filtered terms, in descending order, and then (c) select top X highest ranked terms to identify significantly over-represented functional categories.

### Molecular Function
```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[1]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[1], yaxis= yaxis, margin= list(t= 100))

fig
```

The top negatively expressed gene ontologies from GO molecular function 2018 with significant p-values are vinculin binding and platelet-derived growth factor binding. Vinculin is involved in embryonic development and cell adhesion (1). Platelet-derived growth factor (PDGF) has already been discussed in the context of the kinase receptors for it, and it is involved in development of various tissues and wound healing.
Collagen-binding and integrin binding are again probably involved in the ECM.

1. Peng et al.(2011) New insights into Vinculin function and regulation. International review of cell and molecular biology. https://doi.org/10.1016/B978-0-12-386043-9.00005-0


### Cellular Component
```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[2]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[2], yaxis= yaxis, margin= list(t= 100))

fig
```

The top negatively expressed gene ontologies from GO cellular component 2018 with significant p-values are endoplasmic reticulum lumen, focal adhesion, and phagocytic vesicle membrane. The ER lumen ontology may be related to the ER stress mechanisms already discuessed. Focal adhesion and phagocytic vesicle membrane again related to the ECM and phagosomes.


### Biological Process
```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[3]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[3], yaxis= yaxis, margin= list(t= 100))

fig
```

Again, more ECM relationships are described in the ontologies referring to antigen presentation and ECM organisation. A new topic of an opic-related ontology can be found in the retinal ganglion cell axon guidance.


### Human Phenotype
```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[4]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[4], yaxis= yaxis, margin= list(t= 100))

fig
```

The top negatively expressed gene ontologies from GO cellular component 2018 with significant p-values are endoplasmic reticulum lumen, focal adhesion, and phagocytic vesicle membrane.The ER lumen ontology may be related to the ER stress mechanisms already discuessed. Focal adhesion and phagocytic vesicle membrane again related to the ECM and phagosomes.



## Pathways {.tabset .tabset-fade}

<br>
The top 9 results from each of the databases of interest relating to molecular pathways are displayed below.
<br>

### BioPlanet 2019
```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[5]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[5], yaxis= yaxis, margin= list(t= 100))

fig
```

The top negatively expressed gene pathways from BioPlanet 2019 coexpressed with significant p-values are mostly involved in ECM functions. Perhaps of specific interest:

TWEAK, which is involved in apoptosis (1).Platelet amyloid precursor protein pathways, which is for a protein found in platelets and associated with Alzheimer disease (2). Integrins in angiogenesis, because several other links to angiogenesis have been found earlier in this report.

1 https://pubmed.ncbi.nlm.nih.gov/9405449/

2 https://pubmed.ncbi.nlm.nih.gov/28611024/


### WikiPathway 2021 Human
```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[6]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[6], yaxis= yaxis, margin= list(t= 100))

fig
```

The top negatively expressed gene pathways from Wikipathways 2019 with significant p-values are miRNA targets in ECM and membrane receptors, miR-509-3p alteration of YAP1/ECM (Extracellular matrix related), and Senesence and Autophagy in cancer, and platelet-mediated interactions with vascular and circulating cells.

Again, this brings up more ECM/membrane receptor and platelet relationships with the negatively correlated gene set.

###  KEGG 2021 Human
```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[7]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[7], yaxis= yaxis, margin= list(t= 100))

fig
```

The top negatively expressed gene pathways from KEGG 2019 Human with significant p-values are focal adhesion, phagosome, and ECM-receptor interaction. This again suggests more ECM relationships (ex. focal adhesion, ECM-receptor interaction), but introduces specifically phagosomes and lysosomes. There is also more discussion of the ER protein processing, which can be involved in ER stress (1).

1 https://www.genome.jp/kegg-bin/show_pathway?

## Diseases/Drugs {.tabset .tabset-fade}

<br>
The top 9 results from each of the databases of interest relating to diseases and drugs are displayed below.
<br>

### DisGeNET
```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[8]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[8], yaxis= yaxis, margin= list(t= 100))

fig
```

The top negatively expressed genes relation to diseases/drugs from DSigDB were mostly histamine receptor antagonists.

Suloctidil was a calcium agonist which altered blood platelets, but causes hepatotoxicity (1). Pizotifen is a diarylmethane used in migrane treatment by acting as a serotonin agonist and is a histamine antagonist (2). Terfenadine is a diarylmethane prodrug of a histamine H1-receptor antagonist (3). Prenylamine is also diarylmethane formerly used for angina pectoris (4). Chloropyramine is also an antihistamine (5). Triprolidine and promethazine are also an antihistamines (6,7).

1. https://drugs.ncats.io/substance/XV1N1XY17K
2. https://pubchem.ncbi.nlm.nih.gov/compound/Pizotifen
3. https://pubchem.ncbi.nlm.nih.gov/compound/5405
4. https://pubchem.ncbi.nlm.nih.gov/compound/9801
5. https://pubchem.ncbi.nlm.nih.gov/compound/25295
6. https://pubchem.ncbi.nlm.nih.gov/compound/5282443
7. https://pubchem.ncbi.nlm.nih.gov/compound/4927

### PheWeb 2019

```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[9]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[9], yaxis= yaxis, margin= list(t= 100))

fig
```

The top negatively expressed genes relation to diseases/drugs from ARCHS4 IDG Coexpression with significant p-values are mainly related to NEK7 IDG kinase. NEK7 has already been discussed as being an important gene for mitosis.

## Cell Types {.tabset .tabset-fade}

<br>
The top 9 results from each of the databases of interest relating to cell type are displayed below.
<br>

### Human Gene Atlas

```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[10]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[10], yaxis= yaxis, margin= list(t= 100))

fig
```

Using the top negatively expressed gene the top cell line from the Human Gene Atlasy with highly significant p-values is overwhelmingly smooth muscle.

### Descartes Cell Types and Tissue 2021

```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[11]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[11], yaxis= yaxis, margin= list(t= 100))

fig
```

The top negatively expressed genes related to cell lines from Top CCLE Proteomics 2020 with highly significant p-values are mostly CNS (four glioblastoma) and some Kidney and lung. U118MG is an astrocytoma cell line (1). OSCR2 is a renal cell carcinoma line (2).
U87MG is probably a glioblastoma cell line (although there are issues with its identification) (3). KNS81 is also a glioblastoma (4).
NCIH2052 is a pleural mesothelioma cell line (5). A172 and SNU1105 are also a glioblastoma lines (6,7).

1. https://web.expasy.org/cellosaurus/CVCL_0633
2. https://web.expasy.org/cellosaurus/CVCL_1626
3. https://web.expasy.org/cellosaurus/CVCL_0022
4. https://web.expasy.org/cellosaurus/CVCL_2799
5. https://web.expasy.org/cellosaurus/CVCL_1518
6. https://web.expasy.org/cellosaurus/CVCL_0131

## Transcription {.tabset .tabset-fade}

<br>
The top 9 results from each of the databases of interest relating to transcription are displayed below.
<br>

### ChEA 2016

```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[12]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[12], yaxis= yaxis, margin= list(t= 100))

fig
```

The top negatively expressed transcrpition genes from ChEA_2016 with highly significant adjust p-values are RELA, CJUN, ATF3, and BRD4, within specific cell lines.

RELA codes for part of the most common NK-KB transcription factor, which activates transcription and is involved in many processes and cell types (1). CJUN is a TF, and can be involved in cAMP signaling pathway stimulation and colorectal cancer (2). ATF3 is also involved in cAMP in the response element (3). BRD4 regulates transcription and is involved in epigenetic memory, but is also binds with RELA for NF-KB activation (4). It has been considered as a target for inhibition in leukemia treatment (4).

1. https://www.uniprot.org/uniprot/Q04206#function
2. https://www.uniprot.org/uniprot/P05412#function
3. https://www.uniprot.org/uniprot/P18847#function
4. https://www.uniprot.org/uniprot/O60885#function

### ENCODE and ChEA Consensus TFs from ChIP-X"

```{r,echo=FALSE}
#Part 1 
x <- as.data.frame(enriched[[13]])
x <- x[order(-x$Combined.Score),]
x <- x[1:9,]

#Part 2
fig <- plot_ly(x, x= ~Combined.Score, y= ~reorder(Term,Combined.Score), type = 'bar',marker = list(color=~Adjusted.P.value,size=10, 
              opacity=.9, colorscale='YlGnBu', colorbar=list(title='Adj. P-Value')), textposition = "auto", hoverinfo = "text",
              hovertext = paste(x$Term,"<br> Adj. P-Value:", x$Adjusted.P.value, "<br> Combined Score:", x$Combined.Score))

yaxis <- list(
  zerolinecolor = '#ffff', zerolinewidth = 2, gridcolor = 'ffff',
  title = '', tickmode = "array", margin = list(r = 700))

fig <- fig %>% layout(title = dbs[13], yaxis= yaxis, margin= list(t= 100))

fig
```

The top negatively expressed transcription genes from ChEA_2016 and ENCODE consensus with significant p-values are NFE2L2, SUZ12, and IRF8. NFE2L2 is involved in autophagy and oxidative stress, and is allowed to accumulate during oxidative stress instead of being broken down by the BCR(KEAP1) complex (1).

SUZ12 is part of the PRC2 complex which is involved in repressing transcription, and one of the genes it represses is MYT1 (2). IRF8 can activate or repress transcription, and is part of cell differentiation in CD8+ dendritic cells (WBC) (3).

1. https://www.uniprot.org/uniprot/Q16236#function
2. https://www.uniprot.org/uniprot/Q15022#function
3. https://www.uniprot.org/uniprot/Q02556#function
