<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Cordell Browne" />


<title>ChIP-Seq Analysis</title>

<script src="site_libs/header-attrs-2.13/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<link href="site_libs/vega-1/vega_plots.css" rel="stylesheet" />
<script src="site_libs/vega-1/vega.min.js"></script>
<link href="site_libs/vega-tooltip-1/vega-tooltip.css" rel="stylesheet" />
<script src="site_libs/vega-tooltip-1/vega-tooltip.js"></script>
<script src="site_libs/GlimmaV2-1/MDSSpecs.js"></script>
<script src="site_libs/GlimmaV2-1/XYSpecs.js"></script>
<script src="site_libs/GlimmaV2-1/expressionSpec.js"></script>
<script src="site_libs/GlimmaV2-1/makeTooltip.js"></script>
<script src="site_libs/GlimmaV2-1/saveTools.js"></script>
<script src="site_libs/glimmaMDS-binding-2.4.0/glimmaMDS.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cordell</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">RNA-Seq</a>
</li>
<li>
  <a href="functional.html">Functional Analysis</a>
</li>
<li>
  <a href="ChIP.html">ChIP-Seq</a>
</li>
<li>
  <a href="https://github.com/C0RD3LL/Portfolio">GitHub</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Archive
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="./rnaseqv1/rna-seq.html">RNA-Seq (Original)</a>
    </li>
    <li>
      <a href="./bench/benchmark.html">Benchmark Experiment</a>
    </li>
    <li>
      <a href="./qpcr.html">Wet Lab Research</a>
    </li>
    <li class="dropdown-header">Place Holder 2</li>
    <li class="dropdown-header">Place Holder 3</li>
    <li class="dropdown-header">Place Holder 4</li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">ChIP-Seq Analysis</h1>
<h4 class="author">Cordell Browne</h4>
<h4 class="date"><a href="mailto:cordell@knights.ucf.edu" class="email">cordell@knights.ucf.edu</a></h4>

</div>


<div id="introduction" class="section level1 tabset tabset-fade tabset-pills">
<h1 class="tabset tabset-fade tabset-pills">Introduction</h1>
<div id="objective" class="section level2">
<h2>Objective</h2>
<p>The data for ChIP-seq peak calling are stacks of aligned reads across a genome. Some of these stacks correspond to the signal of interest (e.g. binding of a transcription factor or modified histone). Many other stacks are regarded as molecular or experimental noise, or as being influenced by a systematically greater accessibility of measurement by the experiment at that particular genomic location. The goal of this experiment is to understand how our transcription factors compare to one another in regards to binding sites.</p>
</div>
<div id="about-the-data" class="section level2">
<h2>About the Data</h2>
<p>Source: <a href="https://content.cruk.cam.ac.uk/bioinformatics/software/DiffBind/DiffBind_vignette_data.tar.gz" class="uri">https://content.cruk.cam.ac.uk/bioinformatics/software/DiffBind/DiffBind_vignette_data.tar.gz</a></p>
<p>The easiest way to set up an experiment to analyze is with a sample sheet. The sample sheet can be a , or it can be read directly from a csv file. Here is the example sample sheet read into a from a csv file:</p>
<p>The peaksets are read in using the following DiffBind function:</p>
</div>
</div>
<div id="differential-expression" class="section level1">
<h1>Differential Expression</h1>
<div id="pca" class="section level2">
<h2>PCA</h2>
<p>While the correlation heatmaps already seen are good for showing clustering, plots based on principal components analysis can be used to give a deeper insight into how samples are associated. A PCA plot corresponding to Figure 2, which includes normalized read counts for all 2845 binding sites, can be obtained as follows:</p>
<p>The resulting plot (Figure 5) shows all the MCF7-derived samples (red) clustering on one side of the first (horizontal) component, with the Responsive and Resistant samples not separable either in the first nor in the second (vertical) component.3</p>
<p><em>I recommend to color by “tissue” and shape by “condition”</em></p>
<pre class="r"><code>library(Glimma)
glimmaMDS(dds.tamoxifen, width=800, height=450)</code></pre>
<div id="htmlwidget-6068a4faba7e8e111e2b" style="width:800px;height:450px;" class="glimmaMDS html-widget"></div>
<script type="application/json" data-for="htmlwidget-6068a4faba7e8e111e2b">{"x":{"data":{"mdsData":{"dim1":[-0.512857199285564,-0.590034635563711,1.48050599276988,1.18896196410124,1.28857847707895,-0.910960581260104,-0.964964695810047,1.63157682847133,1.40999971511056,-2.01948472243788,-2.00132114317467],"dim2":[-1.5143143684107,-1.6693832554104,0.776920875741672,0.721027331598259,0.598884356456865,0.702332504294507,0.281839335910233,-0.529746887935494,-0.436410147228023,0.567331087249443,0.501519167733637],"dim3":[0.346937997835066,0.342631694736901,0.697508745184214,0.494478194201397,0.629383286556575,-1.69714509056224,-1.24231219529334,-0.516381911679882,-0.600345324542354,0.813863550520265,0.731381053043406],"dim4":[-0.663815170038446,-0.604410430495303,-0.626179075478736,-0.436369895703423,-0.491499871710906,-0.787328100424526,-0.15924428091844,1.18084663070506,1.10704393620817,0.684599442711454,0.796356815145091],"dim5":[-0.138790285205448,0.0596938030018612,-0.333582019226522,0.398317991973694,0.205479665020088,-0.82456869031011,1.1262395716381,-0.277156559148325,0.020676935316413,-0.29162479517135,0.0553143821115977],"dim6":[-0.0917509597930508,0.123802242912926,-0.736553550214579,0.72175385995441,0.100290108588518,0.226837861810499,-0.383683439842663,-0.0995822876046837,0.118777851108853,-0.040566854935822,0.0606751680155896],"dim7":[-0.0340402353558764,0.0332184893292819,-0.0139935985182401,0.302005332527132,-0.277108584079739,-0.0413688857148968,0.100891414520357,0.665739963423187,-0.725864559345297,0.0646268293467084,-0.0741061661326139],"dim8":[0.187523692450502,-0.134997382416607,0.303095179425128,0.424635425764296,-0.75912369029283,-0.058886123145341,0.0192449753650488,-0.226206176671137,0.253764215070891,0.0450640060581797,-0.0541141216081294],"labels":["BT4741","BT4742","MCF71","MCF72","MCF73","T47D1","T47D2","MCF7r1","MCF7r2","ZR751","ZR752"],"Tissue":["BT474","BT474","MCF7","MCF7","MCF7","T47D","T47D","MCF7","MCF7","ZR75","ZR75"],"Factor":["ER","ER","ER","ER","ER","ER","ER","ER","ER","ER","ER"],"Condition":["Resistant","Resistant","Responsive","Responsive","Responsive","Responsive","Responsive","Resistant","Resistant","Responsive","Responsive"],"Treatment":["Full-Media","Full-Media","Full-Media","Full-Media","Full-Media","Full-Media","Full-Media","Full-Media","Full-Media","Full-Media","Full-Media"],"Peak_caller":["counts","counts","counts","counts","counts","counts","counts","counts","counts","counts","counts"],"Replicate":["1","2","1","2","3","1","2","1","2","1","2"],"sizeFactor":[0.809961019664126,0.823205624684335,0.429899304089376,0.456732255869754,0.578619105836017,0.496227809507736,1.83090873380412,0.765203859609421,0.736158303055218,0.877144536125204,3.1959394477547],"-":["0","0","0","0","0","0","0","0","0","0","0"],"- ":[0,0,0,0,0,0,0,0,0,0,0]},"eigenData":{"name":[1,2,3,4,5,6,7,8],"eigen":[0.41,0.17,0.15,0.12,0.05,0.03,0.02,0.02]},"features":{"numeric":["sizeFactor","- "],"discrete":["labels","Tissue","Factor","Condition","Treatment","Peak_caller","Replicate","-"],"all":["sizeFactor","- ","labels","Tissue","Factor","Condition","Treatment","Peak_caller","Replicate","-"]},"continuousColour":false,"dimlist":["dim1","dim2","dim3","dim4","dim5","dim6","dim7","dim8"]}},"evals":[],"jsHooks":[]}</script>
<p>The bias towards enriched binding in the Responsive case (or loss of binding affinity in the Resistant case) can be visualized using MA and Volcano plots, as shown in the following Section.</p>
<pre class="r"><code>dba.plotVenn(tamoxifen, contrast=1, bDB=TRUE, bGain=TRUE, bLoss=TRUE, bAll=FALSE)</code></pre>
<pre><code>## Generating report-based DBA object...</code></pre>
<p><img src="ChIP_files/figure-html/unnamed-chunk-4-1.png" width="672" /> Venn diagrams are also useful for examining overlaps between peaksets, particularly when determining how best to derive consensus peaksets for read counting and further analysis. Section 8, which discusses consensus peaksets, shows a number of Venn plots in context, and the help page for dba.plotVenn has a number of additional examples.</p>
</div>
<div id="ma-plots" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="tabset tabset-fade tabset-pills">MA Plots</h2>
<p>MA plots are a useful way to visualize the relationship between the overall binding level at each site and the magnitude of the change in binding enrichment between conditions, as well as the effect of normalization on data. An MA plot can be obtained for the Resistant vs Responsive contrast as follows:</p>
<div id="ma-plot-i" class="section level3">
<h3>MA Plot i</h3>
<pre class="r"><code>dba.plotMA(tamoxifen)</code></pre>
<p><img src="ChIP_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
</div>
<div id="ma-plot-ii" class="section level3">
<h3>MA Plot ii</h3>
<pre class="r"><code>plotMA(res)</code></pre>
<p><img src="ChIP_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<p>The plot is shown in Figure 7. Each point represents a binding site, with the 246 points in magenta representing sites identified as differentially bound. There is a blue horizontal line through the origin (0 LFC), as well as a horizontal red curve representing a non-linear loess fit showing the underlying relationship between coverage levels and fold changes. The plot shows how the differentially bound sites appear to have a minimum absolute log fold difference of somewhere between one and two. As we have already seen, it also shows that more ERa binding sites lose binding affinity in the tamoxifen resistant condition than gain binding affinity, as evidenced by more red dots below the center line than are above it. This same data can also be shown with the concentrations of each sample groups plotted against each other plot using dba.plotMA(tamoxifen, bXY=TRUE). Section 7 contains several examples of MA plots, including showing non-normalized data, and the ability to plot any subset of samples against any other set of sample.</p>
</div>
</div>
<div id="volcano-plots" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="tabset tabset-fade tabset-pills">Volcano Plots</h2>
<p>Similar to MA plots, Volcano plots also highlight significantly differentially bound sites and show their fold changes. Here, however, the confidence statistic (FDR or p-value) is shown on a negative log scale, helping visualize the relationship between the magnitude of fold changes and the confidence that sites are differentially bound. For example, the same data as plotted in Figure 7 can be visualized as a volcano plot:</p>
<div id="volcano-plot-i" class="section level3">
<h3>Volcano Plot i</h3>
<pre class="r"><code>library(EnhancedVolcano)</code></pre>
<pre><code>## Loading required package: ggplot2</code></pre>
<pre><code>## Loading required package: ggrepel</code></pre>
<pre><code>## Registered S3 methods overwritten by &#39;ggalt&#39;:
##   method                  from   
##   grid.draw.absoluteGrob  ggplot2
##   grobHeight.absoluteGrob ggplot2
##   grobWidth.absoluteGrob  ggplot2
##   grobX.absoluteGrob      ggplot2
##   grobY.absoluteGrob      ggplot2</code></pre>
<pre class="r"><code>library(magrittr)
  EnhancedVolcano(res ,lab = rownames(res), x = &#39;log2FoldChange&#39;, y = &#39;pvalue&#39;)</code></pre>
<p><img src="ChIP_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
</div>
<div id="volcano-plot-ii" class="section level3">
<h3>Volcano Plot ii</h3>
<pre class="r"><code>dba.plotVolcano(tamoxifen)</code></pre>
<p><img src="ChIP_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<p>The plot is shown in Figure 8, with the predominance of lower binding in the Resistant case evidenced by the greater number of significant sites on the negative side of the Fold Change (X) axis.</p>
</div>
</div>
<div id="heatmaps" class="section level2 tabset tabset-fade tabset-pills">
<h2 class="tabset tabset-fade tabset-pills">Heatmaps</h2>
<p>The effect of different scoring methods (normalization) can be examined in these plots by setting the score parameter to a different value. The default value, DBA_SCORE_NORMALIZED, uses the normalized read counts (see Section 7). Another scoring method is to use RPKM fold (RPKM of the ChIP reads divided by RPKM of the control reads); a correlation heatmap for all the data using this scoring method can be obtained by typing dba.plotHeatmap(tamoxifen, score=DBA_SCORE_RPKM_FOLD).</p>
<div id="cell-line-heatmap" class="section level3">
<h3>Cell-Line Heatmap</h3>
<pre class="r"><code>dba.plotHeatmap(tamoxifen)</code></pre>
<p><img src="ChIP_files/figure-html/unnamed-chunk-9-1.png" width="960" /> The resulting plot (Figure 1) shows that while the replicates for each cell line cluster to- gether appropriately, the cell lines do not cluster into groups corresponding to those that are responsive (MCF7, T47D, and ZR75) vs. those resistant (BT474 and MCF7r) to ta- moxifen treatment. It also shows that the two most highly correlated cell lines are the two MCF7-based ones, even though they respond differently to tamoxifen treatment.</p>
</div>
<div id="binding-site-heatmape" class="section level3">
<h3>Binding Site Heatmape</h3>
<p>Another way to view the patterns of binding affinity directly in the differentially bound sites is via a binding affinity heatmap, showing the read scores for some or all of the binding sites. This can be plotted for the example case as follows:</p>
<pre class="r"><code>hmap &lt;- colorRampPalette(c(&quot;red&quot;, &quot;black&quot;, &quot;green&quot;))(n = 13)
readscores &lt;- dba.plotHeatmap(tamoxifen, contrast=1, correlations=FALSE,
                              scale=&quot;row&quot;, colScheme = hmap)</code></pre>
<p><img src="ChIP_files/figure-html/unnamed-chunk-10-1.png" width="960" /> Figure 10 shows the affinities and clustering of the differentially bound sites (rows), as well as the sample clustering (columns). In this case, the (normalized) counts have been row scaled, and a red/green heatmap color palette applied.</p>
</div>
</div>
</div>
<div id="chip-seq-downstream-analysis" class="section level1">
<h1>ChIP-seq downstream analysis</h1>
<pre class="r"><code>peaks.consensus &lt;- dba.peakset(tamoxifen, bRetrieve = T)
peaks.consensus</code></pre>
<pre><code>## GRanges object with 2795 ranges and 11 metadata columns:
##        seqnames            ranges strand |    BT4741    BT4742     MCF71
##           &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##      1    chr18       90841-91241      * |   2.46925   1.21476   0.00000
##      2    chr18     150222-150622      * |  18.51941  15.79192  51.17477
##      3    chr18     150807-151207      * |  13.58090   9.71811   9.30450
##      4    chr18     189192-189592      * |  13.58090  26.72479  46.52252
##      5    chr18     199821-200221      * |  14.81553  19.43621   2.32613
##    ...      ...               ...    ... .       ...       ...       ...
##   2791    chr18 77782651-77783051      * |  13.58090   1.21476   6.97838
##   2792    chr18 77903180-77903580      * |   8.64239  18.22145   4.65225
##   2793    chr18 77955380-77955780      * |  18.51941  10.93287  18.60901
##   2794    chr18 77968226-77968626      * |  64.20062  70.45627 123.28468
##   2795    chr18 77987616-77988016      * |  22.22329   3.64429 148.87207
##            MCF72     MCF73     T47D1     T47D2    MCF7r1    MCF7r2     ZR751
##        &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
##      1   0.00000    0.0000   0.00000  0.546177    0.0000    0.0000  25.08139
##      2  24.08413   46.6628   0.00000  3.277061   22.2163   12.2256  13.68076
##      3  13.13680   13.8260   2.01520  2.730884   32.6710   24.4513   2.28013
##      4  26.27360   65.6736   6.04561  3.277061  107.1610   73.3538  18.24101
##      5   8.75787   10.3695  10.07602 10.923537   15.6821   19.0176  33.06183
##    ...       ...       ...       ...       ...       ...       ...       ...
##   2791   24.0841   3.45651  14.10642   8.19265   6.53421   9.50882   23.9413
##   2792    0.0000   6.91301   8.06081  11.46971   9.14789   4.07521   30.7817
##   2793   13.1368   8.64126  26.19765   7.10030   2.61368   8.15042   34.2019
##   2794   89.7681  79.49962  12.09122  16.38531 130.68413  66.56177   55.8631
##   2795   89.7681  81.22787  14.10642   6.55412  60.11470  23.09286  205.2113
##            ZR752
##        &lt;numeric&gt;
##      1  10.95140
##      2  10.95140
##      3   5.31925
##      4  13.14168
##      5  17.20934
##    ...       ...
##   2791   10.9514
##   2792   17.8351
##   2793   27.2220
##   2794   42.5540
##   2795  146.7487
##   -------
##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</code></pre>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
