<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8" />
        <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
        <meta name="description" content="" />
        <meta name="author" content="" />
        <title>Cordell Browne Portfolio - Main Page</title>
        <link rel="icon" type="image/x-icon" href="assets/favicon.ico" />
        <!-- Core theme CSS (includes Bootstrap)-->
        <link href="styles.css" rel="stylesheet" />
    </head>
    <body id="page-top">
        <!-- Navigation-->
        <nav class="navbar navbar-expand-lg navbar-dark bg-dark fixed-top" id="mainNav">
            <div class="container px-4">
                <a class="navbar-brand" href="index.html">Cordell Browne</a>
                <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarResponsive" aria-controls="navbarResponsive" aria-expanded="false" aria-label="Toggle navigation"><span class="navbar-toggler-icon"></span></button>
                <div class="collapse navbar-collapse" id="navbarResponsive">
                    <ul class="navbar-nav ms-auto">
                        <li class="nav-item"><a class="nav-link" href="portfolio.html">Portfolio</a></li>
                        <li class="nav-item"><a class="nav-link" href="https://github.com/C0RD3LL">GitHub</a></li>
                        <li class="nav-item"><a class="nav-link" href="index.html">Contact</a></li>
                        
                    </ul>
                </div>
            </div>
        </nav>
        <!-- Header-->
        <!-- About section-->
        <section id="about">
            <div class="container px-4">
                <div class="row gx-4 justify-content-center">
                    <div class="col-lg-8">
                        <h2>Portfolio Summary</h2>
                        <p class="lead">This page contains projects that illustrate my capabilities with R, RMarkdown, python, html, css, plotly, rshiny and more. 
                            Currently it is sectioned into two parts, Clinical Analysis and Bioinformatics analysis. 
                        </p>
                        <ul>
                            <li><a href="#clinic">Clinical Analytics</a></li>
                            <li><a href="#bio">Bioinformatics Analytics</a></li>
                            <li><a href="#other">Other...</a></li>
                        </ul>
                    </div>
                </div>
            </div>
        </section>
        <!-- Content section-->
        <section class="bg-light" id="">
            <div class="container px-4">
                <div class="row gx-4 justify-content-center">
                    <div class="col-lg-8">
                        <h1 id="clinic"><b>Clinical Analysis</b></h1>
                        <br>
                        <a href="Insurance.html"><h2 >Insurance Cost Prediction Modeling; EDA + Regression</h2></a>
                        <p class="lead">In this project, my main aim is to show ways to go deep into the data story-telling.
                             I will also work on a model that gives approximation as to what will be the charges of the patients. Determined by what factors
                             influenced the charge of a specific patient. Lastly, we will go step by step to understand the story behind the patients in this
                             dataset only through this way we could have a better understanding of what features will help our model have a closer accuracy to
                             the true patient charge</p>

                        <br>
                        <br>
                        <h1 id="bio"><b>Bioinformatics Analysis</b></h1>
                        <br>
                        <a href="rnaseq.html"><h2>Oral Squamous Cell Carcinoma (OSCC) RNA-Seq Analysis</h2></a>
                        <p>This RNA-Seq workflow aims to identify which genes and biological processes may be important for our condition of interest 
                            (Oral squamous cell carcinoma, metastasis vs no-metastasis). Starting with the generation of the reference genome, the workflow
                             covers how the data was prepared for exploratory data analysis and Differential expression (DE) analysis (DE includes co-expression)
                              and how it is used in those phases as well. The R package, DESeq2 and other supporting ones were an integral part for visualization
                               of the results, helping better understand the characteristics of the data and also which significant differentially expressed genes
                                could be identified.</p>
                        <p><b>Outline of RNA-seq Workflow:</b></p>
                        <li>Preprocessing: Generating a count matrix from self-generated reference genome and sample reads (fastq)</li>
                        <li>Exploratory data analysis: Answering question like, how does normalization and transformation affect the data?</li>
                        <li>Differential Expression: What genes are most regulated, what is statically important?</li>
                        <li>Co-expression Analysis: Assessing, what genes work together in hubs? What are their expression levels?</li>
                        <br>
                        <a href="functional.html"><h2>OSCC Functional Analysis</h2></a>
                        <p>Functional Enrichment analysis methods helps gain insight about the biology underlying a list of genes. In this case genes
                             that are considered signifcantly differentially expressed. The software package, Enrichr was used to analyzed the gene set.
                              Enrichr currently contains a large collection of diverse gene set libraries available for analysis and download. Overall,
                               Enrichr is a comprehensive resource for curated gene sets and a search engine that accumulates biological knowledge for
                                further biological discoveries.
                        </p>
                        <br>
                        <a href="ChIP.html"><h2>ChIP-Seq Analysis of TF ERa in Breast Cancer Cell Line Replicates</h2></a>
                        <p>ChIP-seq was used to analyze five breast cancer cell lines against the transcription factor ERa. Three of these cell lines are
                             responsive to tamoxifen treatment, while two others are resistant to tamoxifen; there are eleven total sequenced libraries.
                              For each sample, there is an associated peakset derived using the MACS peak caller. Only data for chromosome 18 was included
                               in the data. The goal of the analysis is compare binding across samples and cell types.</p>

                         <p>Tamoxifen is a nonsteroidal antiestrogen that has found successful applications for each stage of breast cancer in the treatment
                             of selected patients. Tamoxifen was originally introduced for the treatment of advanced disease in postmenopausal women. The 
                             proven efficacy of tamoxifen and the low incidence of side effects made the drug an ideal agent to test as an adjuvant therapy
                              for women with node-positive breast cancer</p>      
                        <br>
                        <a href="benchmark.html"><h2>Benchmark experiment</h2></a>
                        <p>Benchmarking different tools/method on their abilities to generate fastq indexes. This compares time in seconds and memory usage of
                             the tools over amount of bases in file and file size in bytes.The benchmarking was done bewteen samtools, pyfastx, biopython, and
                              bioperlâ€™s Index::Fastq module. The results concluded that samtools performed faster then the rest, most likely becuase of its high
                               memory usage. It was also noted that bioperl was signifcantly slower to generate an index than others and its memory usage was kept
                                in a narrow range and consisently low (17,000 ~ 18,000) regardless of file size.</p>
                        <h1 id="other"><b>Other</b></h1>
                        <br>
                    </div>
                </div>
            </div>
        </section>
        <!-- Contact section-->
        <!-- Footer-->
        <footer class="py-5 bg-dark">
            <div class="container px-4"><p class="m-0 text-center text-white">&copy; Cordell Browne 2022</p></div>
        </footer>
        <!-- Bootstrap core JS-->
        <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/js/bootstrap.bundle.min.js"></script>
        <!-- Core theme JS-->
        <script src="js/scripts.js"></script>
    </body>
</html>
