<!DOCTYPE html>
<html>
<head>
<title>Quantitative Real-Time Polymerase Chain Reaction</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
<link href="style.css" rel="stylesheet">
<link rel="preconnect" href="https://fonts.googleapis.com"> 
<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin> 
<link href="https://fonts.googleapis.com/css2?family=Fredoka:wght@500&display=swap" rel="stylesheet">
<a href="https://github.com/C0RD3LL/Portfolio"><i class="fa fa-github" style="font-size:36px"></i></a>
<a href="https://c0rd3ll.github.io/portfolio-0.01/"><i class="fa fa-home" style="font-size:36px"></i><br></a>
</head>

<body>
    <div class="content content1">
            <h1>
                Verification of Housekeeping Genes in Pituitary Adenomas Using
                RT-qPCR for Further Down Stream Analysis.
            </h1>
            <p>Cordell Browne <a href="mailto: cordell@knights.ucf.edu"><i class="fa fa-envelope"></i></a>
            &nbsp &nbsp Dr. Diane White <a href="mailto: Diane.White.ucf.edu"><i class="fa fa-envelope"></i></a></p>
            <a><p>Uni of Central Florida</p></a>

            <div class="dropdown">
                <button class="dropbtn">Sections</button>
                  <div class="dropdown-content">
                    <a href="introduction">Introduction</a>
                    <a href="objectives">Objectives</a>
                    <a href="procedures">Procedures</a>
                    <a href="results">Results</a>
                    <a href="discussion">Discusion</a>
                  </div>
              </div>

              <p><blockquote><h2>Abstract</h2>
                Quantitative real-time polymerase chain reaction (qPCR) is the most reliable tool for gene expression studies.
                Selection of housekeeping genes (HKGs) that are having most stable expression is critical to carry out accurate
                gene expression profiling. There is no 'universal' HKG having stable expression in all kinds of tissues under 
                all experimental conditions.<br>
              </blockquote></p><br>
              
              <a id="introduction"><h2>Introduction</h2></a>
              <p>
                Gene expression analysis provides insight into complex regulatory networks which will lead to the identificati
                on of genes involved in new biological processes or implicated in disease development. Quantitative real-time 
                polymerase chain reaction (qPCR) is often used as a tool to determine the expression of target genes, which are
                normalized against a stable reference gene. qPCR has increasingly gained importance due to its high sensitivity,
                accuracy, reproducibility, and its potential for high throughput [1,2]. The validity of gene expression data 
                determined by qPCR is dependent on the optimal selection of reference genes that are characterized by high 
                stability and low expression variability [3,4,5]. Ideal reference genes should be stably expressed in all tissues or
                cells under investigation irrespective of tissue type, disease state, developmental stage or experimental 
                treatment [6,7]. However, several reports have indicated that variation in the expression levels of endogenous 
                reference genes may occur as a result of pathological change particularly in clinical samples associated with 
                malignant diseases [6,8]. Thus, the selection of appropriate reference genes for clinical patient samples is vital 
                to gene expression analysis. Several publications have shown that Beta-actin and GAPDH vary considerably and are
                thus unsuitable references for RNA transcription analysis [9,10,11,12]. Many studies make use of these reference 
                genes without proper validation of their presumed stability of expression. Convenient housekeeping genes (HKGs) have
                to be verified prior to qPCR data analysis for each tissue type under investigation and each experimental setup used
                [13,14,15]. So far, normalization of qPCR data in glioblastomas (GBM) has mostly been performed with a single reference
                gene. The necessity to use a combination of at least two validated reference genes for accurate quantitative evaluation 
                has not yet been considered [16]. It has been strongly suggested that more than one stably expressed HKG should be used 
                to prevent misinterpretation of gene expression data [11,17,18].
                <br><br>
                Recently the traditional approach to cancer therapy has shifted towards personalized medicine [19]. Gene expression 
                analysis is increasingly being used in combination with clinical variables to diagnose and model prognosis and response
                to therapy [20]. Despite intensive medical efforts, there is a lack of definite information regarding the etiology of
                malignant gliomas. Therefore, it is important to understand the complex biological interactions that regulate glioma
                development. GBM, the highest-grade malignant astrocytoma [21] are the most common and lethal type of tumors in the central
                nervous system. The mean survival time after diagnosis is one year [22]. This poor prognosis is due to their infiltrative
                nature and their resistance to cytotoxic treatments [23]. Many therapies based on characterized genetic alterations are in
                the clinical trials phase, but their efficacy is still below expectations [19]. Thus, there is a need for novel therapeutic
                targets for GBM treatment. In this regard, several studies are dedicated to explore expression profiles, using qPCR to
                identify novel genes differentially expressed in GBM. The present work is aimed to evaluate the suitability of selected HKGs
                for expression analysis in GBM.
                <br><br>
                The purpose of this study was to validate the expression of reference genes suitable for the normalization of qPCR gene
                expression data in GBM. To quantify differentially expressed genes in GBM, it is necessary to find a set of genes with constant
                expression across all samples. GAPDH, HPRT, and TBP are the most commonly used genes for normalization of qPCR data in glioma 
                research. We investigated the expression profile and validated the stability of six HKGs (Table 1) involved in different cellular
                functions, and widely used as normalization genes in the literature for qPCR analysis in human GBM. For the first time we studied
                gene expression profiles of six HKGs from formalin fixed paraffin embedded (FFPE) GBM samples using qPCR. Our study demonstrated
                that FFPE samples retained important prognostic information and thus supports the use of FFPE sections for gene profiling in GBM.
                We determined that TBP and RPL13A are the most suitable reference genes for expression studies in GBM. These results will thus 
                allow analysis of gene expression data in GBM research.<br>
              </p>

              <a id="objectives"><h2>Objectives</h2></a>
              <p>
                ASBMB lists a number of desirable learning goals associated with understanding and manipulation of DNA and RNA, as well
                as experimental design and data analysis. These learning goals are foundational concepts under the category of “Information
                Storage and Transfer.”23 Our observable learning objectives for this research module were that the student will be able to
                demonstrate understanding of the molecular biology, plan and design an experiment based on primary literature, develop a 
                procedure that meets assigned goals, execute their experimental plan in a collaborative fashion, and analyze and communicate
                the results.
              </p>
              <a id="procedures"><h2>Procedures</h2></a>
              <p>
                <h3>4.1 Material</h3>
                
                Materials needed for the research module included synthesized oligonucleotides and primers, typical molecular biology reagents
                and equipment and RT-qPCR instrumentation. Students were not given a lab procedure. Instead, for an inquiry lab, they were 
                provided with background information and primary literature (Supporting Information). A sample student protocol is provided 
                (Supporting Information).<br>

                <h3>4.2 Sampling</h3>
                The CDC Diagnostic Panel provides extensive guidance on sampling, handling and preserving samples, primer choice, reagents used,
                and procedure.24 We only followed their procedures loosely. Students could choose a physiological buffer with tetracycline antibiotic.
                Swabs were dipped in buffer, wiped twice over a surface, then broken off into the collection tube and capped. Collected samples were
                stored at -80°C until use.<br>

                <h3>4.4 Gel electrophoresis</h3>
                Electrophoresis of the positive control was run on a 2.5% agarose gel with the positive control generated by templateless PCR, negative
                 control, and a ladder. The positive control PCR product was purified from the gel slices using a spin-column gel extraction kit according
                to manufacturer's instructions.<br>

              </p>
              <a id="results"><h2>Results</h2></a>
              <p>
                ASBMB lists a number of desirable learning goals associated with understanding and manipulation of DNA and RNA, as well
                as experimental design and data analysis. These learning goals are foundational concepts under the category of “Information
                Storage and Transfer.”23 Our observable learning objectives for this research module were that the student will be able to
                demonstrate understanding of the molecular biology, plan and design an experiment based on primary literature, develop a 
                procedure that meets assigned goals, execute their experimental plan in a collaborative fashion, and analyze and communicate
                the results.
              </p>
              <a id="discusion"><h2>Disscusion</h2></a>
              <p>
                This research module provides upper-level undergraduates the opportunity to design and safely conduct an experiment that 
                combines molecular biology design with a real-world public health application. Experimental procedures that have been previously 
                encountered such as agarose gel electrophoresis are reinforced. The exercise also requires the students to compare published 
                experimental procedures and incorporate them into their own design.The ability to work with DNA and RNA is fundamental for all 
                biologists and biochemists. The use of RT-qPCR is a sensitive technique that is convenient for comparative gene expression. Although
                RNA-seq is becoming more common and more affordable, RT-qPCR is still widely utilized. RT-PCR also has applications in assessing DNA 
                yield and in forensic science.12 Templateless PCR, used to generate the positive control, is a technique whereby large oligos can be 
                designed with overlapping terminal sequences and assembled into a contiguous single-stranded product by PCR. The technique is important
                for synthetic biology and has been used with undergraduates to generate large partially synthetic chromosomal segments.29 The students
                had been warned that we might have no positive environmental samples and that any positive samples would only indicate RNA, not 
                necessarily infectious virus, and not necessarily indicate that there had been enough viral load to be infectious. We did, however, 
                have a sample that had high RNA levels consistent with infectious virus.19 Our findings were almost 2 weeks after sampling in areas 
                that had regular antiviral cleaning, but staff ensured additional decontamination of the area. In addition, to avoid any rumors or 
                concerns from students, administration, or parents, we also created a FAQ about our experiment and findings (Supporting Information).
              </p>
              <a id="reference"><h2>References</h2></a>
              <p>
                1. Ginzinger DG. Gene quantification using real-time quantitative PCR: an emerging technology hits the mainstream.
                Exp Hematol. 2002;30:503–512.<br>
                2. Bustin SA. Quantification of mRNA using real-time reverse transcription PCR (RT-PCR): trends and problems. J Mol
                Endocrinol. 2002;29:23–39<br>
              </p>
    </div>
    
</body>
</html>
