<!DOCTYPE html>
<html>
<head>
<title>qPCR experiment</title>
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
<link href="style.css" rel="stylesheet">
<link rel="preconnect" href="https://fonts.googleapis.com"> 
<link rel="preconnect" href="https://fonts.gstatic.com" crossorigin> 
<link href="https://fonts.googleapis.com/css2?family=Fredoka:wght@500&display=swap" rel="stylesheet">
<a href="https://github.com/C0RD3LL/Portfolio"><i class="fa fa-github" style="font-size:36px"></i><br></a>
</head>
<body>
    <div class="content content1">
            <h1>
                Utilizing polymerase chain reaction and real-time
                polymerase chain reaction to test for SARS-CoV-2 RNA
            </h1>
            <p>Cordell Browne <a href="mailto: cordell@knights.ucf.edu"><i class="fa fa-envelope"></i></a></p>
            <a><p>University of Central Florida</p></a>

            <div class="dropdown">
                <button class="dropbtn">Sections</button>
                  <div class="dropdown-content">
                    <a href="introduction">Introduction</a>
                    <a href="objectives">Objectives</a>
                    <a href="procedures">Procedures</a>
                    <a href="results">Results</a>
                    <a href="discussion">Discusion</a>
                  </div>
              </div>

              <p><blockquote><h2>Abstract</h2>
                We present a course-embedded undergraduate research module that involves real-time polymerase chain 
                reaction testing for the presence of SARS-CoV-2 in environmental samples. A positive control RNA was
                constructed and two RNA extraction methods were compared and a range of primers were available to 
                compare. Using a combination of published protocols, we assembled a successful project that illustrated
                a topical exercise similar to real-world assay development. The exercise is aimed at upper-level undergraduates
                and requires 3 weeks of laboratory periods. The students were able to design and test experimental protocols, 
                while learning about RNA detection. This project could be utilized in upper-level classes including molecular 
                biology, biochemistry, biotechnology, or for independent research projects.<br>
              </blockquote></p><br>
              
              <a id="introduction"><h2>Introduction</h2></a>
              <p>
                Since March 11, 2020, when the World Health Organization declared COVID-19 a pandemic, the entire globe has 
                been scrambling for answers about the virus SARS-CoV-2 (short for severe acute respiratory syndrome coronavirus 
                2). As of October 11, 2020, over 36 million cases of COVID-19 and 1 million deaths have been reported.1 Much of 
                the scientific research community's efforts have been directed toward finding the cheapest, fastest, and most 
                effective way to detect a positive case of COVID-19.2, 3 Effective testing allows for contact tracing and appropriate 
                quarantining measures, which could curtail the spread of the virus.4 Undergraduate biochemistry students at Franklin 
                Pierce University sought to perform their own COVID-19 tests on environmental samples from the Rindge campus. 
                This three-week laboratory module provides instruction in molecular tools such as real-time polymerase chain 
                reaction (RT-PCR) as well as the process by which a clinical diagnostic test can be designed and executed.
                <br><br>
                This laboratory experiment follows the format of a course-based research experiences (CURE). Courses in 
                biochemistry and molecular biology are enriched by course-based research experiences. The U.S. federal “Engage 
                to Excel” report recommended widespread adoption of active learning through problem-based learning such as CURE 
                laboratories5 (Gates, 2012). CUREs have been shown to increase learning,6 increase persistence in science majors7, 
                8 as well as having positive impacts on appreciation for science and career aspirations.9 Some research even supports
                the benefits of CUREs when the research goals are not met.10
                <br><br>
                RT-qPCR is a technique whereby RNA is converted to complementary DNA (cDNA) by reverse transcriptase enzyme, followed 
                by quantitative PCR (qPCR). As with standard PCR, qPCR involves amplification of DNA using cycles of denaturation, 
                annealing, and elongation. However, with qPCR, the presence of a fluorescent dye that binds double-stranded DNA (dsDNA) 
                allows for quantitation. The greater the amount of RNA, the more cDNA created and the fewer doubling cycles before the 
                fluorescence can be detected.11 Thus, a smaller number of cycles to detect fluorescence, the more RNA in the original 
                sample. A number of articles have been published that provide undergraduate exercises using qPCR or RT-PCR. While some 
                of these sources use the acronym RT-PCR to stand for real-time PCR,12, 13 others use RT-PCR to stand for reverse 
                transcription PCR14 and some just refer to the process as qPCR.15 Some COVID testing protocols use the term RT-qPCR,16 
                but our procedure is not a truly quantitative PCR (qPCR) because we did not calculate starting RNA calculations nor 
                use a housekeeping gene for comparison of gene expression; thus, we will refer to the technique as RT-PCR. In our 
                experiment, students investigated the presence of SARS-CoV-2 RNA on high-touch surfaces by swabbing them and performing
                 RT-PCR on the samples.<br>
              </p>

              <a id="objectives"><h2>Objectives</h2></a>
              <p>
                ASBMB lists a number of desirable learning goals associated with understanding and manipulation of DNA and RNA, as well
                as experimental design and data analysis. These learning goals are foundational concepts under the category of “Information
                Storage and Transfer.”23 Our observable learning objectives for this research module were that the student will be able to
                demonstrate understanding of the molecular biology, plan and design an experiment based on primary literature, develop a 
                procedure that meets assigned goals, execute their experimental plan in a collaborative fashion, and analyze and communicate
                the results.
              </p>
              <a id="procedures"><h2>Procedures</h2></a>
              <p>
                <h3>4.1 Material</h3>
                
                Materials needed for the research module included synthesized oligonucleotides and primers, typical molecular biology reagents
                and equipment and RT-qPCR instrumentation. Students were not given a lab procedure. Instead, for an inquiry lab, they were 
                provided with background information and primary literature (Supporting Information). A sample student protocol is provided 
                (Supporting Information).<br>

                <h3>4.2 Sampling</h3>
                The CDC Diagnostic Panel provides extensive guidance on sampling, handling and preserving samples, primer choice, reagents used,
                and procedure.24 We only followed their procedures loosely. Students could choose a physiological buffer with tetracycline antibiotic.
                Swabs were dipped in buffer, wiped twice over a surface, then broken off into the collection tube and capped. Collected samples were
                stored at -80°C until use.<br>

                <h3>4.4 Gel electrophoresis</h3>
                Electrophoresis of the positive control was run on a 2.5% agarose gel with the positive control generated by templateless PCR, negative
                 control, and a ladder. The positive control PCR product was purified from the gel slices using a spin-column gel extraction kit according
                to manufacturer's instructions.<br>

              </p>
              <a id="results"><h2>Results</h2></a>
              <p>
                ASBMB lists a number of desirable learning goals associated with understanding and manipulation of DNA and RNA, as well
                as experimental design and data analysis. These learning goals are foundational concepts under the category of “Information
                Storage and Transfer.”23 Our observable learning objectives for this research module were that the student will be able to
                demonstrate understanding of the molecular biology, plan and design an experiment based on primary literature, develop a 
                procedure that meets assigned goals, execute their experimental plan in a collaborative fashion, and analyze and communicate
                the results.
              </p>
              <a id="discusion"><h2>Disscusion</h2></a>
              <p>
                This research module provides upper-level undergraduates the opportunity to design and safely conduct an experiment that 
                combines molecular biology design with a real-world public health application. Experimental procedures that have been previously 
                encountered such as agarose gel electrophoresis are reinforced. The exercise also requires the students to compare published 
                experimental procedures and incorporate them into their own design.The ability to work with DNA and RNA is fundamental for all 
                biologists and biochemists. The use of RT-qPCR is a sensitive technique that is convenient for comparative gene expression. Although
                RNA-seq is becoming more common and more affordable, RT-qPCR is still widely utilized. RT-PCR also has applications in assessing DNA 
                yield and in forensic science.12 Templateless PCR, used to generate the positive control, is a technique whereby large oligos can be 
                designed with overlapping terminal sequences and assembled into a contiguous single-stranded product by PCR. The technique is important
                for synthetic biology and has been used with undergraduates to generate large partially synthetic chromosomal segments.29 The students
                had been warned that we might have no positive environmental samples and that any positive samples would only indicate RNA, not 
                necessarily infectious virus, and not necessarily indicate that there had been enough viral load to be infectious. We did, however, 
                have a sample that had high RNA levels consistent with infectious virus.19 Our findings were almost 2 weeks after sampling in areas 
                that had regular antiviral cleaning, but staff ensured additional decontamination of the area. In addition, to avoid any rumors or 
                concerns from students, administration, or parents, we also created a FAQ about our experiment and findings (Supporting Information).
              </p>

    </div>
    
</body>
</html>
